src/generic/MapWithInverse.jl

```diff
@@ -61,11 +67,17 @@ retraction_map(f::MapCache) = retraction_map(f.map)
 
 (f::MapWithRetraction{D, C})(a) where {D, C} = (f.map)(a)::elem_type(C)
 
-function Base.show(io::IO, ::MIME"text/plain", M::MapWithRetraction)
-   io = pretty(io)
-   println(io, "Map with retraction")
-   println(io, Indent(), "from ", Lowercase(), domain(M))
-   print(io, "to ", Lowercase(), codomain(M), Dedent())
+function Base.show(io::IO, M::MapWithRetraction)
+   if get(io, :supercompact, false)
+      # no nested printing
+      print(io, "Map with retraction")
+   else
+      # nested printing allowed, preferably supercompact
+      io = pretty(io)
+      print(io, "Map: ")
+      print(IOContext(io, :supercompact => true), Lowercase(), domain(M), " -> ")
+      print(IOContext(io, :supercompact => true), Lowercase(), codomain(M))
```

Comment:
Note how this is now shared by many map implementations (except for "Map:" vs. "Hom:"). Only the "supercompact" thing differs. Could be used to further reduce code duplication and hence uniformity